src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -401,6 +401,22 @@ private TypeGenerationSpec ParseTypeGenerationSpec(in TypeToGenerate typeToGener
                 {
                     classType = ClassType.Enum;
                 }
+                else if (IsMemoryType(type, out INamedTypeSymbol? namedMemoryType, out collectionType))
+                {
+                    underlyingType = namedMemoryType.TypeArguments[0];
+
+                    if (underlyingType.SpecialType == SpecialType.System_Byte)
+                    {
+                        classType = ClassType.BuiltInSupportType;
+                        collectionType = default;
+                        primitiveTypeKind = JsonPrimitiveTypeKind.ByteArray;
```

Comment:
I still need to verify that the fast path code is indeed picking up this flag. Any advice on how validate that the code emitted is correct is greatly appreciated.

Reply:
I don't think you need to set a `JsonPrimitiveKind` in this case, since it's only used for types that have overloads in the `Utf8JsonWriter` class:

https://github.com/dotnet/runtime/blob/d59af2cf097acb100ad6a9cba652be34be6f4a2e/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs#L877C21-L877C76